source/slang/slang-ir-explicit-global-context.cpp

```diff
@@ -12,8 +12,108 @@ namespace Slang
 // thread-group, and wrap them up in an explicit "context"
 // type that gets passed between functions.
 
+enum class HoistableTypes : UInt
+{
+    GlobalVariable = 0b1,
+    GlobalParam = 0b10,
+    All = 0xFFFFFFFF,
+};
+
+enum class HoistGlobalVarOptions : UInt
+{
+    PlainGlobal = 0b0,
+    SharedGlobal = 0b1,
+    RaytracingGlobal = 0b10,
+    All = 0xFFFFFFFF,
+};
+
 struct IntroduceExplicitGlobalContextPass
 {
+    class ExplicitContextPolicy
+    {
+    public:
+        ExplicitContextPolicy(CodeGenTarget target)
+        {
+            switch (target)
+            {
+            case CodeGenTarget::SPIRV:
+            case CodeGenTarget::SPIRVAssembly:
+                hoistableTypes = HoistableTypes::GlobalVariable;
+                requiresFuncTypeCorrectionPass = true;
+                addressSpaceOfLocals = (AddressSpace)SpvStorageClassFunction;
```

Comment:
You may want to leave a TODO comment here so that we can clean up this line after the issue #4742 is resolved.